src/_guides/environment-declarations.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Configuring apps with environment declarations
+description: >
+  Learn about using environment declarations to customize application behavior.
+---
+
+Dart supports specifying environment declarations when
```

Comment:
Consider calling it a "compilation environment".

The word "environment" is overloaded, and traditionally refers to the OS environment, which is available as `Platform.environment`.

The naming of these constructors as `fromEnvironment` has been a constant pain since day 1, so anything we can do to distinguish their meaning from "the other environment" (aka "the real environment") is positive.

Reply:
Thanks for all the great feedback! This is the last remaining suggestion which I'm still working on.

It makes a lot of sense to make it clear these are tied to a "compilation environment", but "compilation environment declarations" is rather long to call these multiple times across the page (and site). So I'm going to give some more thought to how to adjust each mention.

Reply:
I would change the title to "Configuring apps with _compilation_ environment declarations" and clarify the distinction right in the first paragraph with an extra sentence.

> Dart supports specifying compilation environment declarations when
> building or running an application.
> Compilation environment declarations allow you to specify
> configuration values or options as key-value pairs
> that are accessed and evaluated at compile time.
> This page uses "environment" and "environment declarations"
> to refer to the **compilation environment**, not the OS environment
> traditionally referred to as "the environment" in software terminology.

(or "colloquially" instead of "traditionally" in the last line)

Reply:
I made those adjustments, instead putting the extra sentence as a note. Thanks for the idea :)